cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -116,7 +119,7 @@ Result<compute::Declaration> FromProto(const substrait::Rel& rel,
         } else {
           return Status::NotImplemented(
               "substrait::ReadRel::LocalFiles::FileOrFiles::format "
-              "other than FILE_FORMAT_PARQUET");
+              "other than FILE_FORMAT_PARQUET and not recognized");
```

Comment:
```suggestion
              "other than FILE_FORMAT_PARQUET is not recognized");
```

Reply:
What I meant by "and not recognized" is that the given file format (which is specified as other than parquet) cannot be recognized (automatically using the file extension). I'll make the error message clearer.